Sources/XcodeProj/Objects/Project/PBXProject.swift

```diff
@@ -562,16 +564,17 @@ extension PBXProject: PlistSerializable {
             dictionary["packageReferences"] = PlistValue.array(finalPackageReferences)
         }
 
-        var plistAttributes: [String: Any] = attributes
+        var plistAttributes: [String: ProjectAttribute] = attributes
 
         // merge target attributes
-        var plistTargetAttributes: [String: Any] = [:]
+        var plistTargetAttributes: [String: [String: ProjectAttribute]] = [:]
         for (reference, value) in targetAttributeReferences {
-            plistTargetAttributes[reference.value] = value.mapValues { value in
-                (value as? PBXObject)?.reference.value ?? value
```

Comment:
This was added as a feature for `XcodeGen` to be able to easily work with these values.  This complicates support for strong types though and the migration due to the removal of this change would be to set the `.string(object.reference.value)` instead of just `object` at the callsites.

Reply:
The lates updates now include `.targetReference` 

```swift
pbxProject.setTargetAttributes(
    ["TestTargetID": .targetReference(target)], 
    target: testTarget 
)
```